src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/CompressedSparseTensor.cs

```diff
@@ -96,7 +96,7 @@ internal CompressedSparseTensor(Array fromArray, bool reverseStride = false) : b
 
                 foreach (T item in fromArray)
                 {
-                    if (!item.Equals(Zero))
+                    if (!item!.Equals(Zero))
```

Comment:
Are we missing some easier qualification? Maybe just an `Assert(item is not null)` at the start of the foreach?

Reply:
We can do that if you like.  The need for this will go away when https://github.com/dotnet/csharplang/issues/3214 is fixed.

Reply:
I was thinking it would be simpler than introducing so many `!`, but there are several other `foreach` with just one case, rather than multiple

I don't think it should be blocking, I just got tripped up by how `nullable` and generics work again 😄